### PR TITLE
IGAPP-812: Custom logging function

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
@maxammann what do you think of this?

File: build-configs/tools/manage.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (maxammann) [inline build-configs/tools/manage.ts:1]: I think we should disable this rule for the whole tools directory

File: native/src/redux/reducers/__tests__/morphContentLanguage.spec.ts [hunk 2]
```diff
@@ -12,14 +12,23 @@ import {
...
 import { MorphContentLanguageActionType, PushCategoryActionType, PushEventActionType } from '../../StoreActionType'
 import createCityContent from '../createCityContent'
 import morphContentLanguage from '../morphContentLanguage'
 import pushCategory from '../pushCategory'
 import pushEvent from '../pushEvent'
 
+jest.mock('../../../utils/helpers', () => ({
+  forEachTreeNode: jest.requireActual('../../../utils/helpers').forEachTreeNode,
+  log: jest.fn()
+}))
```
[the remark this answers is not in the document]
Comment (maxammann) [reply native/src/redux/reducers/__tests__/morphContentLanguage.spec.ts:26]: but this is no e2e test

File: native/src/sagas/__tests__/loadCityContent.spec.ts [hunk 3]
```diff
@@ -29,6 +29,7 @@ import loadPois from '../loadPois'
 
 jest.mock('../../utils/helpers', () => ({
   ...jest.requireActual('../../utils/helpers'),
+  log: jest.fn(),
   reportError: jest.fn()
 }))
```
Comment (maxammann) [inline native/src/sagas/__tests__/loadCityContent.spec.ts:34]: I think we should mock this by default or not mock this at all. Why are we mocking log and reportError?

File: native/src/utils/helpers.ts [hunk 4]
```diff
@@ -9,13 +9,75 @@ import { FetchError, NotFoundError } from 'api-client/src'
 import buildConfig from '../constants/buildConfig'
 import AppSettings from './AppSettings'
 
+const sentryEnabled = (): boolean => buildConfig().featureFlags.sentry
+const developerFriendly = (): boolean => buildConfig().featureFlags.developerFriendly
+
+export const initSentry = (): void => {
+  if (!sentryEnabled()) {
```
Comment (maxammann) [inline native/src/utils/helpers.ts:16]: Do we require mocking because of sentry? can we avoid mocking somehow?

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
> @maxammann what do you think of this?

looks good!

File: native/src/sagas/__tests__/loadCityContent.spec.ts (hunk 3 above)
Comment (klinzo) [reply native/src/sagas/__tests__/loadCityContent.spec.ts:34]: `reportError` needs to be mocked/spied to be able to assert on it. Removed mocking of `log`

File: native/src/utils/helpers.ts (hunk 4 above)
Comment (klinzo) [reply native/src/utils/helpers.ts:16]: No, I just mock when I want to assert that the function was actually called.

**Pull request merged**